File: CS/Q139166/MySimpleButtonViewInfo.cs
using DevExpress.XtraEditors.ViewInfo;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Drawing;
using DevExpress.XtraEditors.Controls;
using System.Reflection;
using DevExpress.Utils;
namespace DXSample {
    public class MySimpleButtonViewInfo : SimpleButtonViewInfo {
        public MySimpleButtonViewInfo(SimpleButton owner) : base(owner) { }

        protected override EditorButtonObjectInfoArgs CreateEditorButtonObjectInfoArgs(EditorButton editorButton, AppearanceObject appearance) {
            return base.CreateEditorButtonObjectInfoArgs(new MyEditorButton(), appearance);
        }
    }

    public class MyEditorButton : EditorButton {
        public MyEditorButton() : base() { }

        public override System.Drawing.Image Image {
            get { return base.Image; }
            set {
                if (base.Image == value) return;
                FieldInfo fi = typeof(EditorButton).GetField("_image", BindingFlags.Instance | BindingFlags.NonPublic);
                fi.SetValue(this, value);
                OnChanged();
            }
        }
    }
}

File: VB/Q139166/MySimpleButtonViewInfo.vb
Imports Microsoft.VisualBasic
Imports DevExpress.XtraEditors.ViewInfo
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.Drawing
Imports DevExpress.XtraEditors.Controls
Imports System.Reflection
Imports DevExpress.Utils
Namespace DXSample
	Public Class MySimpleButtonViewInfo
		Inherits SimpleButtonViewInfo
		Public Sub New(ByVal owner As SimpleButton)
			MyBase.New(owner)
		End Sub

		Protected Overrides Function CreateEditorButtonObjectInfoArgs(ByVal editorButton As EditorButton, ByVal appearance As AppearanceObject) As EditorButtonObjectInfoArgs
			Return MyBase.CreateEditorButtonObjectInfoArgs(New MyEditorButton(), appearance)
		End Function
	End Class

	Public Class MyEditorButton
		Inherits EditorButton
		Public Sub New()
			MyBase.New()
		End Sub

		Public Overrides Property Image() As System.Drawing.Image
			Get
				Return MyBase.Image
			End Get
			Set(ByVal value As System.Drawing.Image)
				If MyBase.Image Is value Then
					Return
				End If
				Dim fi As FieldInfo = GetType(EditorButton).GetField("_image", BindingFlags.Instance Or BindingFlags.NonPublic)
				fi.SetValue(Me, value)
				OnChanged()
			End Set
		End Property
	End Class
End Namespace